internal/os/system/api.go

```diff
@@ -37,3 +50,51 @@ func (APIImplementor) GetBIOSSerialNumber() (string, error) {
 	}
 	return lines[1], nil
 }
+
+func (APIImplementor) GetService(name string) (*ServiceInfo, error) {
+	script := `Get-Service -Name $env:ServiceName | Select-Object DisplayName, Status, StartType | ` +
+		`ConvertTo-JSON`
+	cmd := exec.Command("powershell", "/c", script)
+	cmd.Env = append(os.Environ(), fmt.Sprintf("ServiceName=%s", name))
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("error querying service name=%s. cmd: %s, output: %s, error: %v", name, cmd, string(out), err)
+	}
+
+	var serviceInfo ServiceInfo
+	err = json.Unmarshal(out, &serviceInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	return &serviceInfo, nil
+}
+
+func (APIImplementor) StartService(name string) error {
+	script := `Start-Service -Name $env:ServiceName`
```

Comment:
Is there a reason to set the `servicename` parameter through an environment variable here? Why not use `sprintf` directly here rather than a couple of lines below?

Reply:
Hey, thanks for the review! :)

I've built this upon the comments that were left in the smb api group:

https://github.com/kubernetes-csi/csi-proxy/blob/69666768532484464908e227950c9fff6f199389/internal/os/smb/api.go#L55-L60

I tried to find more data online about Powershell command injection, but I haven't found anyone referring to env variables. I did find this [link](https://devblogs.microsoft.com/powershell/protecting-against-malicious-c) which might indicate that env variables could prevent injection. 
Anyway, if this really solves injection then it's a good idea to add this in all csi-proxy commands

Reply:
Thanks for pointing out the link! Makes complete sense to use this protection in case the csi-proxy is being used to invoke arbitrary commands in a malicious context on the host.